Multiple use block validation logic improvement

delowardev opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
Fixes: https://github.com/WordPress/gutenberg/issues/38502

## Why? 
Any block that doesn't support the `multiple` feature, doesn't display the **Multiple blocks** validation warning when it's inside group blocks.

## How?
Include `innerBlocks` in the validation process.

## Testing Instructions

1. Open post or page to edit
2. Insert "Group" block
3. Insert the "More" block inside the "Group" block
4. Duplicate "Group" block.

The "Read more" should display the validation warning.

## Screenshots or screencast <!-- if applicable -->

Before:

https://user-images.githubusercontent.com/17360543/160422035-1111351a-8e70-45f4-b9c9-571cc0fd3bed.mov

After:

https://user-images.githubusercontent.com/17360543/160421847-bcc4d749-3e38-43e4-955e-35e258fe6534.mov

delowardev committed:
Multiple block validation logic improvement

getdave reviewed (commented):
Thank you for raising this PR 👏 

I tested it and it works well for the simple case you have described. However I think I may have uncovered some more complex use cases:

- Using within a Query Loop in the Site Editor, if I duplicate the loop then the editor shows an error. Presumably it's valid to use the `More` block twice when it's part of a template so we're going to have to make the validation logic more flexible:
<img width="1510" alt="Screen Shot 2022-04-06 at 10 22 52" src="https://user-images.githubusercontent.com/444434/161942680-41200677-e829-4663-80d0-9454f7a67fee.png">
- I noticed converting the blocks into a Reusable Block circumvents the validation entirely:
<img width="1175" alt="Screen Shot 2022-04-06 at 10 20 39" src="https://user-images.githubusercontent.com/444434/161942233-57dade94-6e09-4a8e-b37a-7e27687ca3fd.png">


In addition:
- I tried with the [`core/read-more` block](https://github.com/WordPress/gutenberg/blob/ae657f38d491895930cd943f2cfa43cef228e7e4/packages/block-library/src/read-more/block.json) and it seems to support `multiple`. I wonder if that's a mistake?
- It would be good if we could add [an e2e test](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/testing-overview.md#end-to-end-testing) to validate this functionality for both nested and non-nested blocks. Probably in `https://github.com/WordPress/gutenberg/tree/trunk/packages/e2e-tests/specs/editor/plugins`

Lastly, just a small note that we try to avoid merge commits and instead rebase against _upstream_ `trunk`. You should do that if you have any conflicts with the _upstream_ Gutenberg `trunk`. Update your `trunk` branch from the upstream main Gutenberg repo and then whilst you have your branch checked out run `git rebase trunk` to rebase.

ntsekouras committed:
Add `edit` context to patterns REST controllers (#40259)

* Add `edit` context to patterns REST controllers

* remove extra check

gziolo committed:
chore(release): publish

 - @wordpress/block-directory@3.4.1
 - @wordpress/block-editor@8.5.1
 - @wordpress/block-library@7.3.1
 - @wordpress/blocks@11.5.1
 - @wordpress/core-data@4.4.1
 - @wordpress/customize-widgets@3.3.1
 - @wordpress/e2e-tests@4.0.1
 - @wordpress/edit-post@6.3.1
 - @wordpress/edit-site@4.3.1
 - @wordpress/edit-widgets@4.3.1
 - @wordpress/editor@12.5.1
 - @wordpress/format-library@3.4.1
 - @wordpress/reusable-blocks@3.4.1
 - @wordpress/server-side-render@3.4.1
 - @wordpress/widgets@2.4.1

Mamaduka committed:
List View: Indicate locked state in block label (#40270)

fabiankaegy committed:
remove WordPress 5.9 specific warning from FSE documentation (#40274)

Mamaduka committed:
i18n: Add context to 'None' strings (#40273)

andrewserong committed:
Cover: Always allow transform from Group block, add handling for Row and Stack variations (#40212)

* Cover: Always allow transform from Group block, add handling for Row and Stack variations

* Move background color or gradient settings to parent, preserve Group block for all variations

talldan committed:
Migrate document settings test to playwright (#40258)

* Add ported version of site editor document settings test

* Delete old puppeteer test

* Fix test title, not a template part being tested

* Use the spec suffix

* Rename test description

* Remove the EditorTitle utility class

* Remove unnecessary deletion of templates and template parts

* Improve selectors

* use toHaveText

* Select the iframe using its title

talldan committed:
Add @talldan as a codeowner for wordpress/preferences (#40295)

kevin940726 committed:
Try to fix flaky playwright tests not being reported (#40290)

gziolo committed:
Plugin: Move the logic for iframed editor to WP 6.0 compat (#40278)

jasmussen committed:
Remove redundant block name from "settings" panels (#40275)

* File block

* Table block.

* Image block.

* Image block native.

* Gallery block.

* Audio block.

* Media & text

* Media & text native

* Video

* Video native

* Spacer

* Archives

* Categories.

* Latest comments.

* RSS

* Tag Cloud

* Post excerpt.

* Author

* Login

fabiankaegy committed:
Doc: Add documentation for `ancestor` property in block metadata (#40027)

* Add documentation for `ancestor` property

Adds documentation for the new API added in https://github.com/WordPress/gutenberg/pull/39894

* Update docs/reference-guides/block-api/block-metadata.md

Co-authored-by: Greg Ziółkowski <grzegorz@gziolo.pl>

* Add ancestors example in innerblocks tutorial

Co-authored-by: Greg Ziółkowski <grzegorz@gziolo.pl>
Co-authored-by: Luis Herranz <luisherranz@gmail.com>

Mamaduka committed:
Block Toolbar: Fix lock icon focus (#40302)

SiobhyB committed:
[RNMobile] Update "add block" button style in default editor view (#39726)

This PR seeks to make the block inserter more visible in the "default view" of the editor. For the purposes of this PR, "default view" has been interpreted as the view of the editor when no blocks (including the editor's title) are selected.

ntsekouras committed:
[Multi-selection]: Copy/cut partial selected blocks (#40098)

* [Multi-selection]: Copy/cut partial selected blocks

* handle `copy` button in block actions

* merge blocks on cut

* partial copy from block actions only when is mergeable

* update selector

* revert partial selection copy in block actions menu item

* add jsdoc for mapRichTextSettings util

* update tests

* add new e2e tests + pressKeyTimes playwright util

Commit:
[Mobile] - Quote block V2 support (#40133)

* Mobile - Update Quote block to support V2

* Mobile - React Native editor tests - Update to pick the first main block list wrapper

* Mobile - Add Quote block V2 experimental flag

* Mobile - React native editor test - Revert block list change, this will be needed once the quote block v2 is enabled by default

* Mobile - Block library - registerCoreBlocks - Make param optional

* Mobile - Edit post test - Update beforeAll call

* Mobile - BlockQuotation - Restore removed code

Mamaduka committed:
Conditionally set secondary sidebar landmark region labels (#40296)

* Conditionally set secondary sidebar landmark region labels
* Don't mutate the object

oandregal committed:
Do not render global styles coming from the server in the site editor (#40185)

zutigrm committed:
Update `theme.json` docs for theme authors (#40005)

Co-authored-by: André <583546+oandregal@users.noreply.github.com>

jsnajdr committed:
useSelect: add async mode test suite and fix two bugs (#40321)

* Test async mode

* Catch missed updates always synchronously in effect

* Rerun mapSelect when registry changes

Commit:
Add README for Gradient and NavigableToolbar (#25580)

glendaviesnz committed:
Add a debounce to the custom color onChange event (#40285)

Co-authored-by: Glen Davies <glen.davies@a8c.com>

scruffian committed:
Elements: Add styles to the footer before the block is rendered (#37728)

* Add styles to the footer before the block is rendered

* Update PHPDoc

* Fix php unit test - length of id was changed by this PR

* Fix linting errors and elements unit tests

* add removed code

* consolidate code from rebase

* Try adding a unit test

* fix test

* Do not short-circuit block rendering

* Rename for clarity

* Do not output styles if skipSerialization is true

* Remove unnecessary test

Co-authored-by: Glen Davies <glen.davies@a8c.com>
Co-authored-by: André <583546+oandregal@users.noreply.github.com>

alexstine committed:
edit-post: After closing welcome and content is empty, focus title field (#40195)

* Posts: Make sure to focus title after welcome dialog close if empty.

* Use a ref to control focus on welcome close.

* Add E2E test coverage.

* Code optimizing. Try to fix E2E tests.

* Reviewer feedback.

* Move isCleanNewPost to edit-post useEffect.

* Only pull initial value of isCleanNewPost.

cbravobernal committed:
Fix comments no results placeholder not appearing (#40234)

* Fix comments no results placeholder not appearing

* Revert changes and apply fix recommended

* Add e2e test for no results

* Comment cleaning

aaronrobertshaw committed:
Borders: Use new border control components in block support (#37770)

This updates border block support to utilise the new `BorderBoxControl` component and offer support for individual side border configuration.

Co-authored-by: Ramon <ramonjd@users.noreply.github.com>
Co-authored-by: André <583546+oandregal@users.noreply.github.com>

Mamaduka committed:
useBlockLock: Always check inherited 'templateLock' status (#40263)

youknowriad committed:
Fixing the flaky template editor mode test (#40325)

* Try fixing the template editor mode test

* Update selector

Co-authored-by: Kai Hao <kevin830726@gmail.com>

aristath committed:
Split navigation block view script (#40135)

* Split modal script in navigation's viewScript

* Enqueue modal script when needed

* revert navigation block changes

* Split navigation block view script

gutenbergplugin committed:
Bump plugin version to 13.0.0

mashikag committed:
Embed block: use preview html for flickr embed previews in editor (#40187)

* Embed block: use embed url over thumbnail_url for html of photo type embeds

This commit is fixing an issue where flickr embeds display pixelated preview images in the block editor.

The issue was that flickr embed (variation of embed block) preview used thumbnail image, over better quality preview options, such as full image url. The thumbnail image is low resolution and 1:1 aspect ratio (often cropped out) version of the original image. Then the preview is stretched to 100% width of the parent element. This resulted in low resolution image being stretched out beyond its width and height. Hence it appeared pixelated.

The issue is fixed by using full image `url`, over `thumbnail_url`, when generating html for `photo` type embed previews.

* Update packages/block-library/src/embed/util.js

Co-authored-by: Nik Tsekouras <ntsekouras@outlook.com>

SiobhyB committed:
Display expanded button when title is selected (#40341)

Display expanded inserter button when title is selected.

youknowriad committed:
Fix focus loss happening when installing blocks from the directory (#40340)

gutenbergplugin committed:
Update Changelog for 13.0.0

Mamaduka committed:
i18n: Fix incorrect '_n' usage (#40292)

* i18n: Fix incorrect '_n' usage
* Use single string for templates

mauriac committed:
Add new option to show or hide empty categories (#39426)

* Packages[categories-block]: Add new option to show or hide empty categories.

Fixes #35489

* Update fixture for unit tests

* Fix inverted behaviour

When "Show empty categories" is checked empty categories do not show
and vice versa. This post fix that behavious.

* Use tab for indentation

youknowriad committed:
Move list and quote block registration to experimental blocks (#40350)

roseg43 committed:
Media & Text: Remove font size declaration from template (#40329)

Co-authored-by: Gabriel Rose <gabriel.rose@10up.com>

michalczaplinski committed:
Add `comment_class()` call to the Comment Template block (#40330)

Co-authored-by: Bernie Reiter <ockham@raz.or.at>

ockham committed:
Post Comments Form: Fix 'Cancel reply' alignment

ockham committed:
Revert "Post Comments Form: Fix 'Cancel reply' alignment"

This reverts commit 50643653f1c08f498705036de565cf39e8c159dd.

cbravobernal committed:
Comments Query Loop block: Add comment ID to all comments (#40268)

For feature parity with previous WordPress Comments markup (prior to FSE). [This WP Tavern post](https://wptavern.com/new-comment-related-blocks-arriving-with-wordpress-6-0#:~:text=The%20first%20issue%20I%20ran%20into%20was%20the%20missing%20comment%20ID%20in%20the%20front%2Dend%20output.) pointed out the issue with the new Comment Query Loop block:

> The first issue I ran into was the missing comment ID in the front-end output. This is necessary for the user’s browser to jump back to their comment after submitting one via the form. I also suspect this is necessary for the comment-reply JavaScript to work when clicking the reply link.

Co-authored-by: Bernie Reiter <ockham@raz.or.at>

SiobhyB committed:
Mobile Release v1.74.0 (#40351)

* Release script: Update react-native-editor version to 1.74.0

* Release script: Update with changes from 'npm run core preios'

* Update CHANGELOG

youknowriad committed:
Fix focus loss when creating pages from link control search results (#40342)

ntsekouras committed:
[Multi-selection]: Copy whole block when selection is collapsed (#40374)

* [Multi-selection]: Copy whole block when selection is collapsed

* rename var to shouldHandleWholeBlocks

youknowriad committed:
Remove async useEffect (#40373)

youknowriad committed:
Fix focus loss when navigating the guide component (#40324)

Mamaduka committed:
Post Featured Image: Use translatable post type label (#40378)

mchowning committed:
[RNMobile] Upgrade to AztecAndroid 1.5.7 (#40099)

* Update aztec to reference 1.5.6 PR

* Update Aztec to 1.5.7

dmsnell committed:
docgen: Unwrap wrapped selectors when inferring types of JSDoc params (#40236)

In #40025 we ran into an obstacle when transitioning code into TypeScript,
namely that our `docgen` tool is unable to find the associated types for
parameters which are documented in JSDoc comments if the parameters are
for the returned function from a call to some wrapping function.

In this patch we're adding two special cases for selectors that call
`createSelector` and `createRegistrySelector` to allow our `docgen` tool
to analyze those inner functions which represent the actual selector.

Fundamentally we should be asking TypeScript for the inferred types of
the function and its parameters but given that we don't have a current
mechanism to do that this issue remains a blocker for broader TypeScript
work. Because of that we're introducing hard-coded special cases for
these common selector wrappers so that we can unblock the TypeScript
work without introducing a generic compromise with potentially-harmful
side-effects, such as might happen if we were to always return the
first argument of a call expression.

t-hamano committed:
E2E: Cross-platform support for Playwright E2E tests (#39765)

Mamaduka committed:
Docs: Update individual block locking section (#40372)

* Docs: Update individual block locking section

* Update inline comment

kevin940726 committed:
Move Playwright snapshots into the same local folder (#40336)

* Move snapshots into a local __snapshots__ folder

* Remove unnecessary snapshot

youknowriad committed:
Clean BlockMover component and styles (#40379)

oandregal committed:
Revert "Move Playwright snapshots into the same local folder (#40336)" (#40412)

This reverts commit 1fc751c845f3908954bdd05fd536dbb9cdcb02bc.

Commit:
Mobile - Layout - Set a background color for the toolbar (#40266)

scruffian committed:
Webfonts: Fix webfont loading in the Site Editor (#40262)

* Webfonts: Fix webfont loading in the Site Editor

* fix linter

Commit:
Fix duotone rendering in site editor (#37727)

Co-authored-by: Alex Lende <ajlende@gmail.com>
Co-authored-by: Ben Dwyer <ben@scruffian.com>
Co-authored-by: André <583546+oandregal@users.noreply.github.com>

Mamaduka committed:
Patterns API: Add 'inserter' to the schema (#40416)

kevin940726 committed:
Fix outdated playwright snapshot directory (#40422)

* Revert "Revert "Move Playwright snapshots into the same local folder (#40336)" (#40412)"

This reverts commit 33a5d514892c254760b5fceb17e72a9faefa1218.

* Fix outdated snapshot directory

Mamaduka committed:
Patterns: Document 'inserter' argument (#40425)

cbravobernal committed:
Add e2e comments tests to stable, wait for selector on set option (#40405)

aaronrobertshaw committed:
Column: Add border support to column blocks (#39967)

Mamaduka committed:
Block Styles: Remove unnecessary button role and 'onKeyDown' handler (#40427)

* Block Styles: Remove unnecessary button role and 'onKeyDown' handler
* Remove tabindex and add aria-role

Co-authored-by: Ramon <ramonjd@users.noreply.github.com>

torounit committed:
[wp-env]: Add support to override `core` by the `WP_ENV_CORE` environment variable. (#40407)

* allow override `core` by the `WP_ENV_CORE` environment variable.

* override coreSource

EliezerSPP committed:
Image Block: Add window to the optional chaining (#40394)

* Add window to the optional chaining

* Image block: Remove the global wp

Mamaduka committed:
Site Editor: Don't render the secondary sidebar in text mode (#40337)

jasmussen committed:
Fix block toolbar text button padding. (#40438)

peterwilsoncc committed:
Comment meta data isn’t queued for lazy loading. (#40242)

mirka committed:
Stop exporting individual color objects from color values file (#40387)

* Stop exporting individual color objects from color values file

* Add changelog

* Replace in React Native

tomasztunik committed:
Fix comments query loop pagination not respecting Discussion Settings (#40284)

* hide comments pagination in the editor if not enabled

* if comments pagination not enabled fetch max allowed items

+ fix possible error where if someone set more than 100 it would return
zero results in the editor but work ok when rendered by the backend

* update selector to return only used data

Co-authored-by: George Mamadashvili <georgemamadashvili@gmail.com>

* clarify that in editor we are rendering pagination placeholder

this is to enable styling the controls regardless of data and state

* Present a user facing warning if comments pagination was disabled.

This will take place of the pagination placeholder.
+ Remove unneeded comment about comments pagination being a placeholder.

* align disabled comments warning copy with other comments blocks

Co-authored-by: George Mamadashvili <georgemamadashvili@gmail.com>

gziolo committed:
Tests: Update unit tests after changes to `build_comment_query_vars_from_block`

alexstine committed:
Navigation block: After choosing an option from Select Menu, focus after block rerender (#40390)

* Focus the select menu button after selection is made.

* Add E2E test.

* Add code comments to E2E test.

fluiddot committed:
Embed block: Fix inline preview cut-off when editing URL (#35326)

* Use empty classname when handling incoming preview

* Force sandbox render when type prop changes

* Add ignore previous classname option to merged attributes

* Move get merged attributes function to util class

ockham committed:
Post Comments Form: Fix layout quirks (#40359)

Co-authored-by: Jeff Ong <jonger4@gmail.com>

cbravobernal committed:
Create Comments Title block with simple styling (#40419)

Co-authored-by: Michal Czaplinski <mmczaplinski@gmail.com>
Co-authored-by: Bernie Reiter <ockham@raz.or.at>

Mamaduka committed:
List View: Remove unused sub-components (#40448)

* Remove ListViewListItem component

* Remove unused ListViewAppender component

jasmussen committed:
Try: Add new duotone filter icon. (#40435)

* Try: Add new duotone filter icon.

* Update changelog.

* Remove test label

Co-authored-by: Alex Lende <alex@lende.xyz>

jasmussen committed:
Try: Rename Navigation Menus to Navigation. (#40413)

* Try: Rename Navigation Menus to Navigation.

* Add beta label.

Mamaduka committed:
useBlockDisplayInformation: Return anchor for block variations (#40458)

jsnajdr committed:
useSelect: cancel render queue in a more straightforward way (#40433)

* priority-queue: add a cancel method

* useSelect: cancel render queue in a less obfuscated way

Mamaduka committed:
Plugin: Clean up non-dynamic block registration list (#40426)

Mamaduka committed:
DocumentSettingPanel: Remove plugin icon fallback (#40355)

fluiddot committed:
Add constants file to Spacer block (#40446)

michalczaplinski committed:
Comment template: Call `comment_class()` before recursing into child comments (#40455)

Co-authored-by: Bernie Reiter <ockham@raz.or.at>

oandregal committed:
e2e test: verify that user can type right after inserting the quote block (#40466)

t-hamano committed:
Tooling: Fix commands containing dot-slashes in npm scripts (#39739)

* fix npm scripts'changelog'

* fix npm scripts 'build:plugin-zip' 'test-unit:date' 'test:create-block'

youknowriad committed:
Add reusable BlockPopover and BlockPopoverInbetween components (#40441)

ciampo committed:
Add new TypeScript project for unit tests (#39436)

* Add test-specific `tsconfig`

* Convert `Text` unit tests to TypeScript

* Add custom Jest matcher types

* Disable `import/no-extraneous-dependencies` for all TypeScript tests

* Remove `@types/node` from `tsconfig.test.json`

* Update `toMatchStyleDiffSnapshot` type defs

* Change snapshot extension from `js` to `tsx`

* Update toMatchStyleDiffSnapshot docs

* Update test to use toMatchStyleDiffSnapshot

* Use alternative link syntax in the JSDoc comment

* Improve comment wording

* Allow null values to be passed to `toMatchStyleDiffSnapshot`

* Remove jest types reference

* Rename testing-library declarations to gutenberg-test-env

* Change approach from a dedicated test project, to re-using the components' existing project

Co-authored-by: Jon Surrell <sirreal@users.noreply.github.com>

Commit:
Mobile - Notice list - Updates the code with the following: (#40415)

- Removes the shouldStack prop which is not used
- Stack all notices in the same position if there's more than one
- Refactor List component to functional component
- Fixes bug where the notices weren't getting removed
- Replace dimension listener with windowDimensions hook

gziolo committed:
Tests: Fix escaping for error message in create block test (#40473)

gziolo committed:
Plugin: Post WordPress 6.0 Beta 1 cleanup (#40349)

khoipro committed:
Scripts: Exclude BEM class from stylelint @wordpress/scripts (#37531)

* Fix #35485 - exclude BEM class

* Update CHANGELOG.md

Co-authored-by: Greg Ziółkowski <grzegorz@gziolo.pl>

mirka committed:
Components: Unify black color to gray-900 (#40391)

* Add comment for preferred color object

* Change pure blacks to gray-900

* Replace darkGray.primary with gray-900

* Replace darkGray.heading with gray-900

* Add changelog

* Update snapshots

* Fix color on InputControl label

gziolo committed:
Create Block: Fix warnings and errors reported during scaffolding and validation (#40479)

* Prettier Config: Align with the CSS Coding Standards

* Docs: Add entries to changelog files

youknowriad committed:
Remove logic preventing the toolbar from hiding when typing within formats (#40476)

dmsnell committed:
docgen: Replace fixtures with inline parsed code. (#40428)

Previously in the `docgen` package test suite we relied on developers manually creating
a set of files for each test that includes a reference snippet of code, a parsed AST,
and an exports listing. These files weren't linked together in any way.

In this patch we're removing the manual steps and inlining the code snippets inside each
test in the suite. This removes a level of manual work and indirection when maintaining
the tests while leaning more on `docgen`'s `engine` component, which itself does the
actual parsing.

gutenbergplugin committed:
Bump plugin version to 13.1.0-rc.1

gutenbergplugin committed:
Update Changelog for 13.1.0-rc.1

gziolo committed:
Plugin: Remove experimental PWA support for WP-Admin (#40475)

gutenbergplugin committed:
Update changelog files

gutenbergplugin committed:
chore(release): publish

 - @wordpress/a11y@3.7.0
 - @wordpress/annotations@2.7.0
 - @wordpress/api-fetch@6.4.0
 - @wordpress/autop@3.7.0
 - @wordpress/babel-preset-default@6.9.0
 - @wordpress/base-styles@4.4.0
 - @wordpress/blob@3.7.0
 - @wordpress/block-directory@3.5.0
 - @wordpress/block-editor@8.6.0
 - @wordpress/block-library@7.4.0
 - @wordpress/block-serialization-default-parser@4.7.0
 - @wordpress/block-serialization-spec-parser@4.7.0
 - @wordpress/blocks@11.6.0
 - @wordpress/components@19.9.0
 - @wordpress/compose@5.5.0
 - @wordpress/core-data@4.5.0
 - @wordpress/create-block-tutorial-template@2.1.0
 - @wordpress/customize-widgets@3.4.0
 - @wordpress/data-controls@2.7.0
 - @wordpress/data@6.7.0
 - @wordpress/date@4.7.0
 - @wordpress/deprecated@3.7.0
 - @wordpress/docgen@1.21.0
 - @wordpress/dom-ready@3.7.0
 - @wordpress/dom@3.7.0
 - @wordpress/e2e-test-utils@7.3.0
 - @wordpress/e2e-tests@4.1.0
 - @wordpress/edit-post@6.4.0
 - @wordpress/edit-site@4.4.0
 - @wordpress/edit-widgets@4.4.0
 - @wordpress/editor@12.6.0
 - @wordpress/element@4.5.0
 - @wordpress/env@4.6.0
 - @wordpress/escape-html@2.7.0
 - @wordpress/eslint-plugin@12.1.0
 - @wordpress/format-library@3.5.0
 - @wordpress/hooks@3.7.0
 - @wordpress/html-entities@3.7.0
 - @wordpress/i18n@4.7.0
 - @wordpress/icons@8.3.0
 - @wordpress/interface@4.6.0
 - @wordpress/is-shallow-equal@4.7.0
 - @wordpress/keyboard-shortcuts@3.5.0
 - @wordpress/keycodes@3.7.0
 - @wordpress/list-reusable-blocks@3.5.0
 - @wordpress/media-utils@3.5.0
 - @wordpress/notices@3.7.0
 - @wordpress/nux@5.5.0
 - @wordpress/plugins@4.5.0
 - @wordpress/postcss-plugins-preset@3.7.0
 - @wordpress/preferences@1.3.0
 - @wordpress/prettier-config@1.2.0
 - @wordpress/primitives@3.5.0
 - @wordpress/priority-queue@2.7.0
 - @wordpress/react-i18n@3.5.0
 - @wordpress/redux-routine@4.7.0
 - @wordpress/reusable-blocks@3.5.0
 - @wordpress/rich-text@5.5.0
 - @wordpress/scripts@22.5.0
 - @wordpress/server-side-render@3.5.0
 - @wordpress/shortcode@3.7.0
 - @wordpress/style-engine@0.6.0
 - @wordpress/token-list@2.7.0
 - @wordpress/url@3.8.0
 - @wordpress/viewport@4.5.0
 - @wordpress/warning@2.7.0
 - @wordpress/widgets@2.5.0
 - @wordpress/wordcount@3.7.0

kevin940726 committed:
Fix replacing undefined in stripAnsi (#40384)

jasmussen committed:
Improve button spacing in the top right corner of the editor (#40411)

* Adjust post editor.

* Site editor.

* Text label option tweaks.

Mamaduka committed:
Edit Post: Remove unused code (#40499)

* Remove onChangeListener function

* Remove unused AdminNotices component

* Remove unused BlockManagerShowAll component

Mamaduka committed:
Block Editor: Deprecate 'BlockColorsStyleSelector' component (#40502)

Mamaduka committed:
Block Locking: Register the 'lock' attribute on the server (#40468)

* Block Locking: Register the 'lock' attribute on the server
* Update registration method
* Update check

ciampo committed:
Eslintrc: add TypeScript tests and stories to development files (#40504)

fabiankaegy committed:
fix indentation inconsistency in docs for block variations (#40515)

gziolo committed:
Scripts: Remove deprecated `format-js` command (#40512)

* Scripts: Remove deprecated `format-js` command

* Update CHANGELOG

* Update CHANGELOG

* Remove more occurrences of `format-js`

youknowriad committed:
Add visualizers for padding and margin for all blocks (#40505)

Co-authored-by: Miguel Fonseca <miguelcsf@gmail.com>

luisherranz committed:
Add placeholder to Post Comments block (#40484)

ciampo committed:
InputControl: remove default value argument from Storybook (#40410)

* InputControl: remove default `value` argument from Storybook

* CHANGELOG

* Option 1: Show complete manual code snippet in Docs

* Option 2: Controlled story with stubbed props in code snippet

* Restore option 1, add "Controlled" Storybook example instead
of option 2

* Remove controlled story

* Fix JSDoc snippet

* Move changelog entry under the right version heading

* Actually remove controlled story

Co-authored-by: Lena Morita <lena@jaguchi.com>

gziolo committed:
Scripts: Remove broken `lint-md-js` command (#40511)

* Scripts: Remove broken `lint-md-js` command

* Update CHANGELOG file

* Update packages/scripts/README.md

Co-authored-by: Ryan Welcher <ryan.welcher@automattic.com>

Co-authored-by: Ryan Welcher <ryan.welcher@automattic.com>

juhi123 committed:
Migrate Image Size test to Playwright (#40467)

* Migrate Image Size to Playwright

* Remove existing utils

* Remove unnecessary utils and image-size-test.js from e2e-tests

cbravobernal committed:
Remove testid and update tests (#40534)

iansvo committed:
URLInput now always has an ID and accessible label (#40310)

* URLInput now always has an ID and accessible label
- Label initialized to null for simpler falsy comparisons
- Duplicate aria-label no longer outputs if a label prop is specified.

luisherranz committed:
Post Comments block: Fix missing label in placeholder (#40527)

Mamaduka committed:
Plugin: Fix fatal error for older WP/PHP versions (#40538)

michalczaplinski committed:
Post Comments Form: Add "proper" visual representation in the editor (#40368)

Co-authored-by: Bruno Ribarić <43731400+ribaricplusplus@users.noreply.github.com>
Co-authored-by: Michael Burridge <mburridge@users.noreply.github.com>
Co-authored-by: David Arenas <david.arenas@automattic.com>
Co-authored-by: Bernie Reiter <ockham@raz.or.at>

ockham committed:
Post Comments Form Block: Mark as stable (#40452)

SantosGuillamot committed:
Add post-comments-form block to comments template (#40256)

ockham committed:
Post Comments block: Enqueue comment-reply script (#40366)

On the frontend, inline the Post Comments block's comment submission form in the position of the "Reply" link that the user clicked.

gziolo committed:
Scripts: Add `--dist-tag` flag to the `packages-update` command (#40514)

* Scripts: Add `--dist-tag` flag to the `packages-update` command

* Update CHANGELOG

ciampo committed:
Global Styles Sidebar: tweak separator margin (#40526)

* Global Styles Sidebar: tweak separator margin and color

* Remove separator color override

pbking committed:
Register all fonts before enqueing any (#40489)

oandregal committed:
Update my notifications (#40549)

dsas committed:
Post terms: Add prefix & suffix

Add prefix and suffix attributes to the post-terms block. These
attributes are displayed inside the block div, immediately before and
after the list. The attributes can be edited as rich text inside the
block editor.

Fixes #29909

dsas committed:
Update integration test for post-terms

aurooba committed:
Docs: Update the Block Metadata title (#40482)

* Update the Block Metadata title

Adding `block.json` to the title makes the document easier to find and identify for folks who may be new to block development and not yet familiar with all the terminology. It would also be good for folks using search engines to find relevant information. :)

* npm run docs:build to update docs manifest

SiobhyB committed:
Use CSS to transform text into title case (#40550)

ciampo committed:
Components: add TypeScript refactor guidelines (#40153)

* Initial draft

* Tiny fixups

* Update packages/components/CONTRIBUTING.md

Co-authored-by: Lena Morita <lena@jaguchi.com>

* Update packages/components/CONTRIBUTING.md

Co-authored-by: Lena Morita <lena@jaguchi.com>

* Add a point regarding `@example` JSDoc on the main export

* Small tweaks

* Add more details to the point about renaming un-connected un-forwarded components

* American english

* Add a point regarding Storybook props table

* Use tabs

* Remove mention of  unstable or experimental props as props to be hidden

* Add more details on how to configure the meta Storybook settings

* Rewrite intro

* Remove "TBD" from unit tests section

Co-authored-by: Lena Morita <lena@jaguchi.com>

carolinan committed:
Update the placeholder text (#40573)

michalczaplinski committed:
Refactor: Reuse code of the editor placeholder across Post Comments and Post Comments Form (#40560)

* Create a separate template for the Comments Form

* Use the form in the Post Comments block

* Add the missing "return" in the CommentForm

t-hamano committed:
Fix: npm script(dev:package) is not working on Windows (#40566)

dsas committed:
Post terms: Wrap prefix & suffix in spans

If the prefix or suffix attributes are provided, wrap them in spans with
a classname.

See #40559

scruffian committed:
Theme Export: Bug: Add .DS_Store to the list of ignored files (#40539)

* Theme Export: Add .DS_Store to the list of ignored files

* change the name of the function

* rename variables

* Update lib/compat/wordpress-6.0/block-template-utils.php

Co-authored-by: Alex Stine <alex.stine@yourtechadvisors.com>

Co-authored-by: Alex Stine <alex.stine@yourtechadvisors.com>

ockham committed:
Comment Template Unit Test: Cover odd/even classes (#40471)

michalczaplinski committed:
Improve the logic for warnings for Post Comments Form placeholder (#40563)

* Improve the warning for Comments Form placeholder

* Fix typo on showPlaceholder variable name

Co-authored-by: Luis Herranz <luisherranz@gmail.com>

glendaviesnz committed:
Add optional chaining in case taxonomy visibility is not defined (#40532)

Co-authored-by: Glen Davies <glen.davies@a8c.com>

gvgvgvijayan committed:
Fixes broken links (#40578)

Fixed markdown format for the links
https://developer.wordpress.org/rest-api/reference/pages/
https://developer.wordpress.org/rest-api/reference/pages/

talldan committed:
Fix template part / reusable block click through on non-safari browsers (#40339)

aaronrobertshaw committed:
Cover: Avoid unnecessary wrapping when transformed from Group (#40602)

Mamaduka committed:
Video: Fix drag and drop upload error state (#40552)

* Video: Fix drag and drop upload error state
* Add missing deps

kevin940726 committed:
Fix incorrect results in the flaky tests reporter (#40567)

* Fix duplicate function name

* Allow more than one line breaks

* Gracefully handle errors when test results is missing

* Refactor and fix dumb mistake again

Commit:
Mobile - RichText - Set a default value for selectionStart / selectionEnd for undefined values (#40581)

talldan committed:
Fix dragging in inspector causing block deselection (#40604)

* Only attempt to select blocks when clicking or highlighting text block text

* Add e2e test

oandregal committed:
Element styles: remove the WordPress filter callback (#40594)

Commit:
Turn off WP 6.0 stopgap handler to use Webfonts API. (#40555)

* Turn off WP 6.0 stopgap handler to use Webfonts API.

* Update load.php

* Update load.php

Co-authored-by: Greg Ziółkowski <grzegorz@gziolo.pl>

carolinan committed:
Fix alignment issue with the comment reply link (#40582)

Removes duplicate font size class.

carolinan committed:
Fix alignment issue with the comment edit link (#40584)

ntsekouras committed:
[Automated Testing]: Fix partial selection copy flaky tests (#40607)

* [Automated Testing]: Fix partial selection copy flaky tests

* add same delay to the other similar tests

carolinan committed:
Fix alignment issue with comment author name (#40610)

* Remove duplicate font size class

* Update unit tests

Co-authored-by: Carlos Bravo <carlos.bravo@automattic.com>

mirka committed:
Global Styles Sidebar: Tweak spacing (#40588)

annezazu committed:
Adding schema info to theme.json reference (#40599)

Received feedback from a developer that including a reference to the theme.json schema would go a long way as it's common for developers to rely on a reference over a how to guide.

jorgefilipecosta committed:
Fix: Show add pattern label when patterns are being prioritised. (#40598)

mirka committed:
Global Styles Sidebar: Rename `NavigationButton` so semantics are clearer (#40590)

* PR 📌 Rename NavigationButton so semantics are clearer

* Bandaid fix semantics in header back button

tomasztunik committed:
Cover: background-attachment fixed iOS bug handling to properly work with iOS 13 (#40087)

Commit:
Mobile Release v1.74.1 (#40606)

* Release script: Update react-native-editor version to 1.74.0

* Release script: Update with changes from 'npm run core preios'

* Update CHANGELOG

* Release script: Update react-native-editor version to 1.74.1

* Release script: Update with changes from 'npm run core preios'

* Mobile - RichText - Set a default value for selectionStart / selectionEnd for undefined values (#40581)

* Mobile - Update changelog

Co-authored-by: Siobhan <siobhan@automattic.com>

fluiddot committed:
[RNMobile] Prevent unnecessary re-renders of RichText component due to color palettes prop (#40615)

dsas committed:
Post terms: Add missing closing span

See #40559

glendaviesnz committed:
Check that a transform matches at the time of running the transform (#40497)

Co-authored-by: Glen Davies <glen.davies@a8c.com>
Co-authored-by: Aaron Robertshaw <60436221+aaronrobertshaw@users.noreply.github.com>

ramonjd committed:
Style Engine: add typography and color to backend  (#40332)

* Initial commit: add typography to style engine

* Adding whitespace to inline output

* Introduce spacing to inline styles
First stab at generating classnames based on slugs

* Removing generating styles based on an options path since we're not using it yet.
Adding a new public function to generate_classnames.
Updated tests.

* Perform kebab case normalization in the style engine.
Remove value_func for common style definitions since I'm pretty sure we'll be using the get_css_rules method for 80%++ of cases.

* Remove unused arg $options

* Refactor style engine so that `generate()` will return an array of css and classnames.
Add mechanism to pass classname slugs
Add typography and color support
Update tests

* Remove 'inline' as an option until we need something different
Remove duplicated classnames from other test
Check for falsy values in style array before we add required classnames

* Only add the required classnames if there's a valid style value or classname slug.

* Removed options arg until we need it. Thanks, linter.

* Refactor generate to accept string CSS preset property values (for non-array values).
Updated tests

* Updated color and typography to use the preset value for the style engine

* No short ternaries says the linter. Booooh!

* Cleaning up. Reinstating deprecated function so we don't forget to deprecate.
Using _wp_array_get where it looks neater.

* Wherefore art thou, vile linter!

* Updating inline docs

* Created public interface method gutenberg_style_engine_generate

* Linter, thy foul stench offends me!

* Using the correct preset property keys for colors and gradients ensures that we set the correct classname.

jostnes committed:
[RNMobile] - Update tests with paragraph blocks to work on both environments and platforms (#40404)

* add fixes for tests with paragraph blocks

* lint fixes

* update missed paragraph blocks to use getTextBlockLocatorAtPosition

* lint fixes

* remove unused code and revert ios coordinate condition

* lint fix

* remove extra backspace press

* rename for clarity

Co-authored-by: jos <17252150+jostnes@users.noreply.github.com>

talldan committed:
Add new preferences persistence API, and save editor preferences in user meta (#39795)

* Implement basic API

* Add new local storage persistence package

* Only call persistenceLayer.set if there is a persistence layer

* Wire up dependencies

* Fix exporting create as default and get not being async

* Remove old persistence interface and configure new persistence layer in post editor

* Use correct user id and storage key

* Move local storage persistence layer into the preferences package

* Try a wp-preferences package

* Use database persistence

* Only set preference values with particular actions

* Remove component for now

* Make native initialize function async

* Revert changes to preferences package.json

* Try using an inline script to configure preferences

* Use a filter to add the necessary script dependencies

* Remove import and package.json dependency from edit-post in favor of wiring up scripts in PHP

* Remove merging of state

* Try abort controller for cancelling multiple in flight requests

* Revert "Try abort controller for cancelling multiple in flight requests"

This reverts commit 06345b57253f80c3619be3e34c6c5b0e2d6aa0b9.

* Try throttling API requests

* Simplify dependencies

* Test that preferences are not persisted in tests due to reload

* Reset all preferences in e2e test setup

* Try `keepalive`

* Oh, this is actually a debounce, not a throttle

* Try a localstorage fallback

* Avoid `null` value becoming preloaded state

* Improve naming and docs

* Revert "Test that preferences are not persisted in tests due to reload"

This reverts commit 3865610409ae2ce960ade361f1e75e213263da11.

* Use a stable reference when storage is empty

* Migrate block editor insert usage to preferences store

Update tests

Make updateInsertUsage a proper action that can be unit tested

Fix reusable block tests

* Also reset preferences after each test

* Use beforeEach for clearing e2e test browser state

* Try avoiding beforeAll and afterAll in nonce test

* Revert "Use beforeEach for clearing e2e test browser state"

This reverts commit e51a00a08bdfba21fcc2d2598d428ef7740d28f2.

* Switch to using beforeEach and afterEach, as they also ensure correct order of operation

* Revert "Migrate block editor insert usage to preferences store"

This reverts commit da21c6793cc7d2beeb084c23260004131c8624bc.

* Migrate basic preferences in from localstorage

* Port the migrate-individual-preference function

* Port the migrate-feature-preference function

* Copy convert-edit-post-panels function from persistence plugin

* Add moveInterfaceEnableItems function and improve folder structure

* Tidy up migration code

* Add a test for the whole migration and fix a bug the test found

* Improve comment

* Fix some more migration issues

* Remove old migration code

* Add some docs for setPersistenceLayer

* Update database persistence layer docs

* Add preferences package changelog entries

* Remove unused local storage persistence folder

* Remove unused local storage persistence layer

* Add unit tests to preferences store

* Add tests for persistence layer

* Handle equality when comparing timestamps

* Improve the preferences README further

* Improve docs

* Move php code into wordpress-6.1 folder

* Reset preferences when running playwright tests

* Fix typo

Co-authored-by: Robert Anderson <robert@noisysocks.com>

* Always register user meta

* Use an IIFE for the inline script and and convert to old school JS

* Use edit context

* Use ISO date for preferences and rename to modified (not a timestamp any more)

* Add clarifying comment about blog prefix

* Use only one underscore

* Make asnync debounce more traditional

* Write tests for debounce async function

* Fix timestamps in persistence layer tests

* Use fake timers

* Split the PHP hooks in two

* Minimize code in inline script

* Rename database-persistence-layer to preferences-persistence

Co-authored-by: Robert Anderson <robert@noisysocks.com>

dependabot[bot] committed:
Bump async from 2.6.1 to 2.6.4 (#40365)

Bumps [async](https://github.com/caolan/async) from 2.6.1 to 2.6.4.
- [Release notes](https://github.com/caolan/async/releases)
- [Changelog](https://github.com/caolan/async/blob/v2.6.4/CHANGELOG.md)
- [Commits](https://github.com/caolan/async/compare/v2.6.1...v2.6.4)

---
updated-dependencies:
- dependency-name: async
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ntsekouras committed:
[RichText]: Ignore selection changes on non contentEditable nodes (#40611)

kevin940726 committed:
Retain trace on failure on CI (#40586)

JustinyAhin committed:
Allow negative values for margin inputs (#40464)

* Enable negative margins in box control

* Use the spacingType property to define when negative margin is accepted

* Use a boolean to allow negative values for margin

aaronrobertshaw committed:
Remove unused BorderStyleControl component (#40635)

kevin940726 committed:
Fix Element.outerHTML (#40638)

chriszarate committed:
ESLint Plugin: Fix `recommended` preset when `prettier` is not installed (#40634)

* Fix `recommended` ESLint preset when `prettier` is not installed

The `prettier` package is incorrectly mandatory when using the `@wordpress/eslint-plugin/recommended` preset. Failure to install it results in an error.

`prettier` is intended to be an optional peer dependency, with related rulesets added only when the user has installed `prettier`. However, the `@worpress/prettier-config` package eagerly loads `prettier`'s `package.json`, leading to an error when it is not installed.

This change moves the require of `@worpress/prettier-config` inside the relevant `if` block, so that it is only loaded when `prettier` is confirmed to be installed.

* Update CHANGELOG.md

Co-authored-by: Greg Ziółkowski <grzegorz@gziolo.pl>

youknowriad committed:
Improve the useDisabled hook and Disabled component (#40631)

SiobhyB committed:
[RNMobile] Add featured image settings to the Latest Posts block (#39257)

This change adds featured image settings to the native Latest Posts block. This is a step towards bringing the available block settings on the native side in alignment with the web.

ndiego committed:
Fix focal point error when the Featured Image is applied (#40643)

Commit:
Docs: Omit name from settings object in registerPlugin (#40629)

youknowriad committed:
Remove the spotlight mode from template parts (#40652)

Mamaduka committed:
Block Locking: Avoid 'lock' attribute registration warning (#40654)

Mamaduka committed:
Patterns: Fix core 'Featured' pattern category registration (#40650)

mirka committed:
PR 📌 Fix header styles and semantics (#40592)

Commit:
Mobile - HTML View - Add a top margin to avoid the tooltip overlapping the title (#40312)

gutenbergplugin committed:
Bump plugin version to 13.1.0

gutenbergplugin committed:
Update Changelog for 13.1.0

noisysocks committed:
Design tweaks to Post Visibility popover (#40530)

* Adjust copy and margins

* Fix mobile styling

* Simplify CSS

* Add close button

* Use hooks

* Adjust colors and margins to look more like mockup

* Anchor visibility popover to middle of entire row

Mamaduka committed:
Audio: Correctly handle drag and drop upload errors (#40623)

aristath committed:
Revert "Allow negative values for margin inputs (#40464)" (#40674)

This reverts commit a109d31c02042ad3c6e055f7aea61de9c999e57b.

Co-authored-by: Justin Ahinon <justiny.ahinon@gmail.com>

kevin940726 committed:
Fix `batchRest` not throwing errors in Playwright (#40668)

peterwilsoncc committed:
[Block] Display paragraph breaks in comment contents block.  (#40667)

* Apply `comment_text` filter in comment template.

* I think this is how I am supposed to test this.

* Fix tests paragraphs break/I broke.

* CS Fix.

* Tidy assertions even more.

* discard discardWhiteSpace — it was a bad idea.

* Thinking face.

jsnajdr committed:
useSelect: prevent nested component update after unmount (#40677)

* useSelect: prevent nested component update after unmount

* Update store name

youknowriad committed:
Remove padding prop from popover (#40544)

mirka committed:
Storybook: Make description-loader more reliable (#40616)

* Storybook: Fix deprecated description path

* Don't overwrite existing parameter config

* Edit for readability

* Prioritize explicit story description

mirka committed:
InputControl: Add flag for larger default size (#40622)

* InputControl: Add flag for larger default size

* Add changelog

jsnajdr committed:
useSelect: improve transition from async to sync mode (#40680)

jasmussen committed:
Try: Fix regression with featured images in latest posts. (#40662)

mirka committed:
UnitControl: Add flag for larger default size (#40627)

* Add prop to types

* Simplify paddingRight override logic

* Add changelog

mirka committed:
Components: Fix error in CONTRIBUTING guide (#40682)

jasmussen committed:
Try: Simpler clear button. (#40437)

* Try: Simpler clear button.

* Update snapshot

* Update snapshot

Co-authored-by: Alex Lende <alex@lende.xyz>

ryelle committed:
Separator Block: Add top & bottom margin support (#40551)

DAreRodz committed:
Comment Content: Show moderation message (#40612)

Show a moderation message in Comment Content right after publishing a comment that needs moderation.

Co-authored-by: Carlos Bravo <carlos.bravo@automattic.com>

cbravobernal committed:
Prevent spam links on author name in comments (#40659)

Do not include the author link regardless of the block settings ([see wp-develop source](https://github.com/WordPress/wordpress-develop/blob/3fdce9eb4b13ab9e9f52c5636b730aaa43740b12/src/wp-includes/class-walker-comment.php#L329-L333)). Issue mentioned in #40601.

ockham committed:
Revert "Prevent spam links on author name in comments (#40659)" (#40701)

This reverts commit 9605f3c07b667099d88f04def050d2bac7a2654b.

ockham committed:
Strip comment author link in unit test

ockham committed:
Revert "Strip comment author link in unit test"

This reverts commit df5fa1bd74d05c02737fe10b6a6046ac65aec639.

mkaz committed:
Silenzio Bruno. Remove @mkaz from CODEOWNERS (#40685)

mauriac committed:
Fix/typos in documentation (#40579)

* Packages[block-editor]: Fix typo in ImageSizeControl doc

* Packages[core-data]: Remove space in __experimentalGetDirtyEntityRecords docblock

talldan committed:
Make debounceAsync return promises that can be caught (#40641)

danieliser committed:
Fix/block meta icon prop errors (#40710)

* Fix hyphenated prop errors
* Correct wrong icon export name

Mamaduka committed:
Navigation Link: Avoid unnecessary re-renders (#40696)

SiobhyB committed:
Mobile Release v1.75.0 (#40630)

Mobile Release v1.75.0.

glendaviesnz committed:
Include text alignment when transforming from paragraph to header and back (#40709)

Co-authored-by: Glen Davies <glen.davies@a8c.com>

Mamaduka committed:
Navigation: Improve selector performance (#40700)

* Update variable names and cleanup

ntsekouras committed:
[Writing Flow]: Try to fix multi-selection with `shift+click` (#40687)

* [Writing Flow]: Try to fix multi-selection with `shift+click`

* change check with collapsed selection

* address feedback

* target specific mouse event with shift

* add e2e tests

gziolo committed:
Preferences: Improve initial configuration for the new persistence package (#40715)

Commit:
TextControl: Covert component to TypeScript (#40633)

Co-authored-by: Marco Ciampini <marco.ciampo@gmail.com>

ellatrix committed:
Quote v2: implement exiting on Enter at end (#39911)

Co-authored-by: Riad Benguella <benguella@gmail.com>
Co-authored-by: André <583546+oandregal@users.noreply.github.com>

ockham committed:
Prevent spam links of author's name in pending moderation comments (take two) (#40702)

* Revert "Revert "Prevent spam links on author name in comments (#40659)" (#40701)"

This reverts commit a06a03d69c23ffc84e25ca946812d27caa163e11.

* Strip comment author link in unit test

oandregal committed:
Add `settings` hook (#40547)

Co-authored-by: Miguel Fonseca <miguelcsf@gmail.com>

Mamaduka committed:
Editor: Refactor PostFeaturedImage component (#40126)

Improve uploading state

Mamaduka committed:
getBlockTransformItems: Support single block object (#40718)

* getBlockTransformItems: Support single block object
* Update navigation link
* Add unit tests

luisherranz committed:
Export hydrate in `@wordpress/element` (#40723)

getdave committed:
Replace gutenberg_render_block_core_navigation_link call with unprefixed render_block_core_navigation_link in the unit tests (#40657)

mauriac committed:
Packages[core-data]: Accept numeric values for the "include" property in getQueryParts (#40570)

* Packages[core-data]: Change "include" type in getQueryParts

Convert the value of the parameter include into a string when it's
a number.

* Add some units test

daviedR committed:
Fix ColorPalette component's documentation (#40620)

The `className` property is documented to have default value of `Select or Upload Media` which is not true, so we should remove it.

cbravobernal committed:
Add no comments count placeholders (#40728)

We have both singular and plural editing on the comment title. Also, we have an option for showing comments count or not.
In singular, this last option was not taken into account, showing always the same placeholder. 'One response'

Also, to keep consistency, we change uppercase or lowercase on R letter of Response depending on we are enabling or disabling the comments count.

carolinan committed:
Update post content placeholder (#40177)

* Updated post content placeholder

* Update the paragraphs

ockham committed:
Comments block: Fix glitches found while backporting (#40628)

cbravobernal committed:
Fix style comment awaiting moderation indentation (#40681)

jorgefilipecosta committed:
Fix: Page patterns don't show when only one pattern is available. (#40707)

carolinan committed:
Update the placeholder for post excerpt (#40178)

* Update the placeholder for post excerpt

* Update edit.js

chad1008 committed:
Components: migrate `ConfirmDialog` component's Stories from knobs to controls (#40164)

ConfirmDialog Stories: simplify stories and convert from knobs to controls

Sisanu committed:
Fixes mobile icon exported constant (#40748)

talldan committed:
Preferences persistence - mark modified property as not readonly in schema (#40637)

* Mark modified property as not readonly

* Remove context field from _modified which is specified on the outer object

peterwilsoncc committed:
Prevent comments rendering on password protected posts. (#40755)

ciampo committed:
Update user-event version to 14.1.1 (#40761)

danieliser committed:
[components] Use internal `Icon` component for `ToggleGroupControlOptionIcon` (#40717)

* Use internal `Icon` component to support dashicons

* Add changelog note

mirka committed:
SelectControl: Finish typescript migration (#40737)

* SelectControl: Move types

* Inherit types from underlying components

* Add named export

* Convert stories to TS and Controls

* Convert tests

* Add changelog

* Simplify size prop types

Co-authored-by: Marco Ciampini <marco.ciampo@gmail.com>

* Use `text` control for prefix/suffix props

Co-authored-by: Marco Ciampini <marco.ciampo@gmail.com>

talldan committed:
Improve tools panel accessibility (#40716)

* Use correct role for reset button

* Use aria-disabled instead of disabled to ensure focus stays on menu item

* Avoid closing dropdown to prevent unexpected focus movement

* Simplify dropdown label

* Add a description to buttons

* Ensure aria-selected is not set on reset button

* Use a single `if` statement

* Use simple string props for roles

* Speak when using menu items

* Make the reset button disabled when there is nothing to reset

* Add an announcement for reset all

* Update changelog

* Use the word options instead of control

* Fix/remove ToolPanel tests

* Simplify button logic and improve tests

* Simplify test selector

* Add test for reset all disabled state

* Add more aria-live tests

* Reintroduce test

Co-authored-by: Ramon <ramonjd@users.noreply.github.com>

talldan committed:
Remove require editor hooks line from full-content test setup (#40772)

ciampo committed:
Tests: restore real timers after using Jest's fake timers (#40760)

* Tests: restore real timers after using Jest's fake timers

* Apply/restore timers before/after ALL instead of EACH for native tests

ntsekouras committed:
[Gradient Tool]: Fix closing of popover when the angle control is clicked (#40735)

getdave committed:
Fix Nav Sidebar Site Editor error (#40722)

getdave committed:
Remove Navigation Area block (#40645)

* Remove all references to Navigation Area block

* Remove rest init

* Remove Nav Area from ts entities

* Remove refs to Nav Areas from Core Data tests

ockham committed:
Comments Query Loop block: Rename to Comments (but keep ID) (#40763)

Co-authored-by: Carlos Bravo <carlos.bravo@automattic.com>
Co-authored-by: Michal Czaplinski <mmczaplinski@gmail.com>

miminari committed:
Migrate paragraph test to Playwright (#40779)

ciampo committed:
UnitControl: rewrite tests in TypeScript (#40697)

* Rename file extensions

* Rewrite `ControlledSyncUnits` part

* Use diff snapshot to test classnames

* Use `toBeInTheDocument`

* Type internal test state setters

* Use `testing-library` semantic queries instead of `document.querySelector`

* Tweak optional parameter for some  utilities

* Add type guard to `hasUnits`

* Fix TS errors in utils tests

* Add types to user-event wrapper function

* CHANGELOG

* Add comment about failing test

* ts-expect-error instead of ts-ignore

* Fix broken test by passing the missing `value`

* Add temporary stories to check single unit behavior

hypest committed:
Update native mobile wrangler for rich-text (#40783)

Trapsta committed:
Close overlay menu when clicking an anchor link  (#39625)

* fix(block-library): close overlay menu in nav block when navigating to an anchor link

Closes #gutenberg-39585

* fix(block-library): add eventlistener to close modal on clicking overlay menu nav links

Closes #gutenberg-39585

* fix(block-library): pass id of modal to micromodal.close and check if overlay menu is open before trying to close it

Closes #gutenberg-39585

* fix(block-library): ignore non-anchor links when closing navigation menu overlay for anchor links

Closes #gutenberg-39585

* chore(block-library): move code for closing modal on clicking anchor links to seperate view-modal.js script

Closes #gutenberg-39585

* fix(block-library): ignore anchor links which open on a new tab when closing modal on clicking link

Closes #gutenberg-39585

* Update packages/block-library/src/navigation/view-modal.js

update comment description

Co-authored-by: Dave Smith <getdavemail@gmail.com>

* Update packages/block-library/src/navigation/view-modal.js

fix comment typo

Co-authored-by: Dave Smith <getdavemail@gmail.com>

Co-authored-by: Dave Smith <getdavemail@gmail.com>

delowardev commented:
@getdave Thanks for the feedback. My apologies for the delayed response.


> _I tried with the [core/read-more block](https://github.com/WordPress/gutenberg/blob/ae657f38d491895930cd943f2cfa43cef228e7e4/packages/block-library/src/read-more/block.json) and it seems to support multiple. I wonder if that's a mistake?_ 

It will be `core/more` block actually; it doesn't support `multiple`
https://github.com/WordPress/gutenberg/blob/bc78b2cf55ca855c728b2beea6489fae56751a53/packages/block-library/src/more/block.json#L23

> _Using within a Query Loop in the Site Editor, if I duplicate the loop then the editor shows an error. Presumably, it's valid to use the More block twice when it's part of a template so we're going to have to make the validation logic more flexible._

I think `core/more` is not supposed to be used in the query loop/post-template; instead, it should be used in the post content.

delowardev committed:
Merge branch 'fix/multiple-block-validation' of github.com:delowardev/gutenberg into fix/multiple-block-validation

delowardev committed:
Multiple use of block validation e2e test

delowardev commented:
Sorry for the inconvenience, my local git history got messed-up
